sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -383,8 +383,7 @@ case class HashAggregateExec(
     val isSupported =
       (groupingKeySchema ++ bufferSchema).forall(f => CodeGenerator.isPrimitiveType(f.dataType) ||
         f.dataType.isInstanceOf[DecimalType] || f.dataType.isInstanceOf[StringType] ||
-        f.dataType.isInstanceOf[CalendarIntervalType]) &&
-        bufferSchema.nonEmpty
```

Comment:
No, I'm sorry, I don't quite understand `bufferSchema.nonEmpty` is for what?

For example, for query `SELECT distinct a, b from tab`  ?

Reply:
Got it. 
> No

`bufferSchema.nonEmpty` was a legacy part of old commit in order to narrow down the scope in other cases. I also think it's not used in that case.
>  I don't quite understand bufferSchema.nonEmpty is for what? For example, for query SELECT distinct a, b from tab ?

Reply:
Hmm, I'm wondering if we have test coverage for this?

Reply:
This condition was added in #12710. It was not explained clearly, but I suppose that it is added because for performance improvement on it. I read through related codes, seems this condition doesn't actually cause issue (I might be missing it out actually). It is better if you can confirm there are test coverage or you can add some.

Reply:
Could you address the above @viirya 's comment about test coverage, @wankunde ?

Reply:
Hi, @viirya @dongjoon-hyun , thanks for your review. 
Add a UT for this change: https://github.com/apache/spark/pull/41685